ios/Ciphers/AesCipher.swift

```diff
@@ -10,124 +10,248 @@ import Foundation
+        assert(bytesOut == Int(encryptedBytes))
+      }
+
     }
-    
-    
-    // Decrypts an input stream to an output stream given a key and an IV
-    // using AES256 CTR mode with No padding
-    
-    public func decrypt(input: InputStream, output:OutputStream, key: [UInt8], iv: [UInt8], callback: (_:RnCryptoError?, _: Status?) -> Void ) -> Void {
-   
-        if !self.isValidIv(iv: iv) {
-            return callback(RnCryptoError.badIv, nil)
+
+    // Final check
+    let status = cryptStream.final(
+      bufferOut: &outputBuffer, byteCapacityOut: outputBuffer.count,
+      byteCountOut: &encryptedBytes)
+
+    // Everything ok, close streams
+    input.close()
+    output.close()
+
+    callback(nil, status)
+
+  }
+
+  // Encrypts an input stream to multiple output streams in chunks
+  public func encryptToChunks(
+    input: InputStream,
+    outputs: [OutputStream],
+    key: [UInt8],
+    iv: [UInt8],
+    chunkSize: Int,
+    callback: (_ error: RnCryptoError?, _ status: Status?) -> Void
+  ) {
+    let algorithm = StreamCryptor.Algorithm.aes
+    let bufferSize: Int = 1024 * 8
+    guard isValidIv(iv: iv) else {
+      return callback(.badIv, nil)
+    }
+
+    let cryptStream = StreamCryptor(
+      operation: .encrypt,
+      algorithm: algorithm,
+      mode: .CTR,
+      padding: .NoPadding,
+      key: key,
+      iv: iv
+    )
+
+    var inputBuffer = [UInt8](repeating: 0, count: bufferSize)
+    var outputBuffer = [UInt8](repeating: 0, count: bufferSize)
+    input.open()
+    var encryptedBytes: Int = 0
+    var chunkIndex = 0
+    var bytesWrittenInChunk = 0
+    var totalBytesProcessed: Int64 = 0
+
+    guard !outputs.isEmpty else {
+      input.close()
+      return callback(.fileCreationFailed, nil)
+    }
+
+    defer {
+      input.close()
+      outputs.forEach { $0.close() }
+    }
+
+    outputs.forEach { $0.open() }
+
+    func writeBytes(from buffer: [UInt8], offset: Int, length: Int) -> (
+      bytesWritten: Int, error: RnCryptoError?
+    ) {
+      if chunkIndex >= outputs.count {
+        return (0, .writeFailed)
+      }
+      let bytesOut = outputs[chunkIndex].write(
+        Array(buffer[offset..<(offset + length)]), maxLength: length)
+      if bytesOut <= 0 {
+        return (0, .writeFailed)
+      }
+      return (bytesOut, nil)
+    }
+
+    while input.hasBytesAvailable {
+      let bytesRead = input.read(&inputBuffer, maxLength: inputBuffer.count)
+      if bytesRead <= 0 { break }
+
+      let status = cryptStream.update(
+        bufferIn: inputBuffer,
+        byteCountIn: bytesRead,
+        bufferOut: &outputBuffer,
+        byteCapacityOut: outputBuffer.count,
+        byteCountOut: &encryptedBytes
+      )
+
+      if status != .success {
+        return callback(.encryptionFailed, status)
+      }
+
+      var offset = 0
+      var remainingBytes = encryptedBytes
+
+      while remainingBytes > 0 {
+        let spaceLeftInChunk = chunkSize - bytesWrittenInChunk
```

Comment:
The `chunkSize` value validation is specially important as if it is 0 or negative, this will break